graphql.go

```diff
@@ -9,25 +9,66 @@ import (
 	"net/http"
 
 	"github.com/shurcooL/graphql/internal/jsonutil"
-	"golang.org/x/net/context/ctxhttp"
 )
 
+func getDefaultClientHeaders() map[string]string {
+	return map[string]string{
+		"Content-Type": "application/json",
+	}
+}
+
+// ClientOptFunc graphql client option
+type ClientOptFunc func(*Client)
+
+// WithHeader set graphql client header
+func WithHeader(key, val string) ClientOptFunc {
+	return func(c *Client) {
+		c.headers[key] = val
+	}
+}
+
+// WithCookie set graphql client cookie
+func WithCookie(key, val string) ClientOptFunc {
+	return func(c *Client) {
+		if c.cookies == nil {
+			c.cookies = map[string]string{}
+		}
+		c.cookies[key] = val
+	}
+}
+
 // Client is a GraphQL client.
 type Client struct {
 	url        string // GraphQL server URL.
 	httpClient *http.Client
+
+	headers,
+	cookies map[string]string
 }
 
 // NewClient creates a GraphQL client targeting the specified GraphQL server URL.
 // If httpClient is nil, then http.DefaultClient is used.
-func NewClient(url string, httpClient *http.Client) *Client {
+func NewClient(url string, httpClient *http.Client) (c *Client) {
```

Comment:
```suggestion
func NewClient(url string, httpClient *http.Client) *Client {
	var c *Client
```
I think you meant to honor the old function signature now, right?